src/amo/reducers/addonsByAuthors.js

```diff
@@ -1,125 +1,139 @@
 |};
 
-type FetchOtherAddonsByAuthorsAction = {|
-  type: typeof FETCH_OTHER_ADDONS_BY_AUTHORS,
-  payload: FetchOtherAddonsByAuthorsParams,
+type FetchAddonsByAuthorsAction = {|
+  type: typeof FETCH_ADDONS_BY_AUTHORS,
+  payload: FetchAddonsByAuthorsParams,
 |};
 
-export const fetchOtherAddonsByAuthors = (
-  { addonType, authors, errorHandlerId, slug }: FetchOtherAddonsByAuthorsParams
-): FetchOtherAddonsByAuthorsAction => {
-  if (!errorHandlerId) {
-    throw new Error('An errorHandlerId is required');
-  }
-
-  if (!slug) {
-    throw new Error('An add-on slug is required.');
-  }
-
-  if (!addonType) {
-    throw new Error('An add-on type is required.');
-  }
-
-  if (!authors) {
-    throw new Error('Authors are required.');
-  }
-
-  if (!Array.isArray(authors)) {
-    throw new Error('The authors parameter must be an array.');
-  }
+export const fetchAddonsByAuthors = (
+  { addonType, authors, errorHandlerId, excludeAddonBySlug }: FetchAddonsByAuthorsParams
+): FetchAddonsByAuthorsAction => {
+  invariant(errorHandlerId, 'An errorHandlerId is required');
+  invariant(addonType, 'An add-on type is required.');
+  invariant(authors, 'Authors are required.');
+  invariant(Array.isArray(authors), 'The authors parameter must be an array.');
 
   return {
-    type: FETCH_OTHER_ADDONS_BY_AUTHORS,
+    type: FETCH_ADDONS_BY_AUTHORS,
     payload: {
       addonType,
       authors,
       errorHandlerId,
-      slug,
+      excludeAddonBySlug,
     },
   };
 };
 
-type LoadOtherAddonsByAuthorsParams = {|
-  slug: string,
+type LoadAddonsByAuthorsParams = {|
   addons: Array<ExternalAddonType>,
+  excludeAddonBySlug?: string,
 |};
 
-type LoadOtherAddonsByAuthorsAction = {|
-  type: typeof LOAD_OTHER_ADDONS_BY_AUTHORS,
-  payload: LoadOtherAddonsByAuthorsParams,
+type LoadAddonsByAuthorsAction = {|
+  type: typeof LOAD_ADDONS_BY_AUTHORS,
+  payload: LoadAddonsByAuthorsParams,
 |};
 
-export const loadOtherAddonsByAuthors = (
-  { addons, slug }: LoadOtherAddonsByAuthorsParams
-): LoadOtherAddonsByAuthorsAction => {
-  if (!slug) {
-    throw new Error('An add-on slug is required.');
-  }
-
-  if (!addons) {
-    throw new Error('A set of add-ons is required.');
-  }
+export const loadAddonsByAuthors = (
+  { addons, excludeAddonBySlug }: LoadAddonsByAuthorsParams
+): LoadAddonsByAuthorsAction => {
+  invariant(addons, 'A set of add-ons is required.');
 
   return {
-    type: LOAD_OTHER_ADDONS_BY_AUTHORS,
-    payload: { slug, addons },
+    type: LOAD_ADDONS_BY_AUTHORS,
+    payload: { addons, excludeAddonBySlug },
   };
 };
 
 type Action =
-  | FetchOtherAddonsByAuthorsAction
-  | LoadOtherAddonsByAuthorsAction;
+  | FetchAddonsByAuthorsAction
+  | LoadAddonsByAuthorsAction;
 
 const reducer = (
   state: State = initialState,
   action: Action
 ): State => {
   switch (action.type) {
-    case FETCH_OTHER_ADDONS_BY_AUTHORS:
+    case FETCH_ADDONS_BY_AUTHORS:
+      if (!action.payload.excludeAddonBySlug) {
+        return state;
```

Comment:
Don't you need to reset the `byUserId` state by making empty arrays for each author that is being fetched? That's what we'd typically do. I think you may need a new `byUserName` mapping for this to work, which I've explained in detail below.

Reply:
Oh shoot, yes, this is totally from an earlier, dumber refactor. But since I'm not using it anywhere yet I didn't catch it.